### PR TITLE
Debug and Launcher implementation

### PR DESCRIPTION
**Pull request by hevayo**

### CODE REVIEW

File: modules/web/js/ballerina/views/function-invocation-view.js [hunk 1]
```diff
@@ -81,6 +81,11 @@ define(['lodash', 'log', './simple-statement-view', './../ast/function-invocatio
             });
             this.listenTo(model, 'update-property-text', this.updateFunctionalExpression);
 
+            this._createDebugIndicator({
+                model: this._model,
```
Comment (pahans) [inline modules/web/js/ballerina/views/function-invocation-view.js]: did the above on this PR https://github.com/ballerinalang/composer/pull/733

File: modules/web/css/ballerina-file-editor.css [hunk 2]
```diff
@@ -114,7 +114,7 @@
 }
 
 .text-editor {
-    height: calc(100% - 80px);
+    height: calc(100%);
```
Comment (dakshika) [inline modules/web/css/ballerina-file-editor.css]: If we are not calculating anything we can remove the calc

File: modules/services/workspace-service/src/main/java/org/wso2/ballerina/tooling/service/workspace/launcher/LaunchManager.java [hunk 3]
```diff
@@ -0,0 +1,262 @@
...
+        BufferedReader reader = new BufferedReader(new InputStreamReader(this.program.getInputStream()));
+        String line = "";
+        try {
+            while ((line = reader.readLine())!= null) {
+                pushMessageToClient(launchSession, LauncherConstants.OUTPUT, LauncherConstants.DATA, line);
+            }
+            pushMessageToClient(launchSession, LauncherConstants.EXECUTION_STOPED , LauncherConstants.INFO,
+                    LauncherConstants.END_MESSAGE);
+        } catch (IOException e) {
+            e.printStackTrace();
```
Comment (mohanvive) [inline modules/services/workspace-service/src/main/java/org/wso2/ballerina/tooling/service/workspace/launcher/LaunchManager.java]: Any reason to use e.printStackTrace here? In above you have used "pushMessageToClient" method and pass the exception there.

File: modules/services/workspace-service/src/main/java/org/wso2/ballerina/tooling/service/workspace/launcher/LaunchManager.java [hunk 4]
```diff
@@ -0,0 +1,262 @@
...
+    }
+
+    public void stopProcess() {
+        int pid = -1;
+        if (this.program != null && this.program.isAlive()) {
+            //todo shutdown implementation
+        }
+    }
+
+
```
Comment (mohanvive) [inline modules/services/workspace-service/src/main/java/org/wso2/ballerina/tooling/service/workspace/launcher/LaunchManager.java]: Unnecessary new line.

File: modules/services/workspace-service/src/main/java/org/wso2/ballerina/tooling/service/workspace/launcher/LaunchServer.java [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+                    //todo enable log in debug mode
+                    //.handler(new LoggingHandler(LogLevel.INFO))
+                    .childHandler(new LaunchServerInitializer());
+            Channel ch = b.bind(port).sync().channel();
+
+            ch.closeFuture().sync();
+            bossGroup.shutdownGracefully();
+            workerGroup.shutdownGracefully();
+        } catch (InterruptedException e) {
+            //@todo proper error handling
```
Comment (mohanvive) [inline modules/services/workspace-service/src/main/java/org/wso2/ballerina/tooling/service/workspace/launcher/LaunchServer.java]: Why error handling defined as "todo" ?

File: modules/services/workspace-service/src/main/java/org/wso2/ballerina/tooling/service/workspace/launcher/LaunchServerHandler.java [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+        if (frame instanceof PingWebSocketFrame) {
+            ctx.channel().write(new PongWebSocketFrame(frame.content().retain()));
+            return;
+        }
+        if (!(frame instanceof TextWebSocketFrame)) {
+            throw new UnsupportedOperationException(String.format("%s frame types not supported", frame.getClass()
+                    .getName()));
+        }
+
+
```
Comment (mohanvive) [inline modules/services/workspace-service/src/main/java/org/wso2/ballerina/tooling/service/workspace/launcher/LaunchServerHandler.java]: Please avoid unnecessary lie brakes.

File: modules/services/workspace-service/src/main/java/org/wso2/ballerina/tooling/service/workspace/launcher/LaunchManager.java [hunk 7]
```diff
@@ -0,0 +1,262 @@
...
+    public static LaunchManager getInstance() {
+        synchronized (LaunchManager.class) {
+            if (launchManagerInstance == null) {
+                launchManagerInstance = new LaunchManager();
+            }
+        }
+        return launchManagerInstance;
+    }
+
+
```
Comment (mohanvive) [inline modules/services/workspace-service/src/main/java/org/wso2/ballerina/tooling/service/workspace/launcher/LaunchManager.java]: Please fix unnecessary line space..

**Pull request merged**